js/components.tsx

```diff
@@ -137,10 +138,14 @@ export const Text = forwardRef<
     >
 >((props, ref) => {
     const context = useContext(MaterialContext);
+    const theme = useContext(ThemeContext);
     const mat = props.material ?? useMemo(() => context.textMaterial?.clone(), []);
     if (mat) {
         (mat as unknown as FlatMaterial).setColor(
-            parseColor(props.color ?? 'fff', tempColor)
+            parseColor(
+                props.color ?? theme.colors?.text ?? (mat as unknown as FlatMaterial).color,
```

Comment:
```suggestion
                props.color ?? theme.colors?.text ?? ((props.material ?? context.textMaterial) as unknown as FlatMaterial).color,
```

Alright, this is a bit of a puzzler 😅 This fixes a very niche bug where someone might have had a color property set, but then based on some UI state reverts it back to the "default", which is the theme or the _original_ material (not the clone).